core_impl/src/main/java/com/google/instrumentation/trace/TraceParams.java

```diff
@@ -0,0 +1,149 @@
+
+  static final TraceParams DEFAULT =
+      TraceParams.builder()
+          .setSampler(DEFAULT_SAMPLER)
+          .setMaxNumberOfAttributes(DEFAULT_SPAN_MAX_NUM_ATTRIBUTES)
+          .setMaxNumberOfAnnotations(DEFAULT_SPAN_MAX_NUM_ANNOTATIONS)
+          .setMaxNumberOfNetworkEvents(DEFAULT_SPAN_MAX_NUM_NETWORK_EVENTS)
+          .setMaxNumberOfLinks(DEFAULT_SPAN_MAX_NUM_LINKS)
+          .build();
+
+  /**
+   * Returns the global default {@code Sampler}. Used if no {@code Sampler} provided in {@link
+   * StartSpanOptions}.
+   *
+   * @return the global default {@code Sampler}.
+   */
+  abstract Sampler getSampler();
+
+  /**
+   * Returns the global default max number of {@link Attributes} per {@link Span}.
+   *
+   * @return the global default max number of {@code Attributes} per {@link Span}.
+   */
+  abstract int getMaxNumberOfAttributes();
+
+  /**
+   * Returns the global default max number of {@link Annotation} events per {@link Span}.
+   *
+   * @return the global default max number of {@code Annotation} events per {@code Span}.
+   */
+  abstract int getMaxNumberOfAnnotations();
+
+  /**
+   * Returns the global default max number of {@link NetworkEvent} events per {@link Span}.
+   *
+   * @return the global default max number of {@code NetworkEvent} events per {@code Span}.
+   */
+  abstract int getMaxNumberOfNetworkEvents();
+
+  /**
+   * Returns the global default max number of {@link Link} entries per {@link Span}.
+   *
+   * @return the global default max number of {@code Link} entries per {@code Span}.
+   */
+  abstract int getMaxNumberOfLinks();
+
+  private static Builder builder() {
+    return new AutoValue_TraceParams.Builder();
+  }
+
+  abstract Builder toBuilder();
+
+  @AutoValue.Builder
+  abstract static class Builder {
+
+    /**
+     * Sets the global default {@code Sampler}.
+     *
+     * @param sampler the global default {@code Sampler}.
+     * @return this.
+     */
+    abstract Builder setSampler(Sampler sampler);
+
+    /**
+     * Sets the global default max number of {@link Attributes} per {@link Span}.
+     *
+     * @param maxNumberOfAttributes the global default max number of {@link Attributes} per {@link
+     *     Span}. It must be positive.
+     * @return this.
+     */
+    abstract Builder setMaxNumberOfAttributes(int maxNumberOfAttributes);
+
+    /**
+     * Sets the global default max number of {@link Annotation} events per {@link Span}.
+     *
+     * @param maxNumberOfAnnotations the global default max number of {@link Annotation} events per
+     *     {@link Span}. It must be positive.
+     * @return this.
+     */
+    abstract Builder setMaxNumberOfAnnotations(int maxNumberOfAnnotations);
+
+    /**
+     * Sets the global default max number of {@link NetworkEvent} events per {@link Span}.
+     *
+     * @param maxNumberOfNetworkEvents the global default max number of {@link NetworkEvent} events
+     *     per {@link Span}. It must be positive.
```

Comment:
Then I read down a dozen lines and saw the checkState calls. Add the appropriate @throw doc everywhere applicable.